core/modules/query/TestFactory.cc

```diff
@@ -67,10 +67,10 @@ void TestFactory::addSelectField(std::shared_ptr<SelectStmt> const& stmt, String
         ValueFactorPtr fact(ValueFactor::newColumnRefFactor(cr));
         ValueExprPtr expr = std::make_shared<ValueExpr>();
         expr->getFactorOps().push_back(ValueExpr::FactorOp(fact));
-        sl->getValueExprList()->push_back(expr);
+        sl.getValueExprList()->push_back(expr);
     }
 
-    stmt->setSelectList(sl);
+    stmt->setSelectList(sl.clone());
```

Comment:
Why do you think this change is better? You create one more instance of SelectList which you then clone, looks like completely unnecessary clone.